### DISCUSSION TITLE
Background Tasks in Blueprints

### DISCUSSION THREAD

**Discussion by butchewing** [pallets/quart#263] [Q&A] [answered]
Is it possible to instantiate a Background Task in a Blueprint route?

```
bp.add_background_task(Data.background_task(id, user_message))
```

This raises the following Error: 'Blueprint' object has no attribute 'add_background_task'

**Comment by davidism** [comment, marked as answer]
It's not specific to a blueprint. Import `current_app` and use that.

```
from quart import current_app

current_app.add_background_task(...)
```
